source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -87,13 +87,17 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
               ProcessingMode::BodySendMode_Name(body_mode_));
     const auto& common_response = response.response();
     if (common_response.has_header_mutation()) {
-      const auto mut_status = MutationUtils::applyHeaderMutations(
-          common_response.header_mutation(), *headers_,
-          common_response.status() == CommonResponse::CONTINUE_AND_REPLACE,
-          filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_,
-          shouldRemoveContentLength());
-      if (!mut_status.ok()) {
-        return mut_status;
+      if (headers_ != nullptr) {
```

Comment:
This header response indicates the ext_proc filter is configured to send headers.  Then the headers_ must not be nullptr.  Do you know in which case the headers_ could be nullptr with such configuration?

In other word, I don't understand why we need to add a headers_ nullptr check when handling header response.  Is it to handle the sidestream server is not trusted, which can send random response case?

Reply:
Yes, as I explained to folks in that internal thread, I don't see that we are seeing null header ptr at this moment.

But I think it is a good safety measure to have. Also forward thinking if further changes accidentally set it to null, we don't want to crash here, which is bottom line.

Reply:
My opinion is we should not just a check here and let it go.
1) If this is to deal with bad code changes, then we should ASSERT(headers_!=nullptr) to capture that.
2) If this is to deal with untrusted side stream server, then we should mark the response as spurious, and close the side stream connection to avoid it make further damage to Envoy.   
As ext_proc has to support untrusted side stream server,  I guess we should do 2).  Also, please add a test for the change.

Reply:
(2) is already done in ext_proc, that is how I explained the folks that this header_ptr cannot be nullptr since out-of-order message has already resulted in spurious message/stream close.

In short , **this PR is not for fixing any error case, it is future safe-proof.**  I don't think we can add a test for this due to reason I mentioned above. 

I can add a assert but it will still be compiled out in production build.

Reply:
@tyxia Yeah, the out-of-order message is already marked as spurious and stream close.  And header_ will never be nullptr here with the current state machine.  Then why we need this check at all?
If the idea of the PR is future proof to avoid someone adding bad code to break the state machine, then probably adding ASSERT() is better to capture the problem early instead of mask it.

Reply:
Yes, @yanjunxiang-google  My purpose is `future proof to avoid someone adding bad code to break the state machine`

 I think @yanavlasov 's suggestion of using `requestHeaders()` is also a good idea. I will revisit this change later as it is not high priority I think.